### PR TITLE
Amended built-in name vars to varibs

### PR DESCRIPTION
**Pull request by jremes-foss**
There was some variable shadowing. Variable name "vars" was actually a shadow built-in. I renamed this variable to "varibs" avoid confusion.

Thank you.

### CODE REVIEW

**Comment by toolforger** [comment]
That's obscuring rather than clearing up what that is.
Alternatives that I can think of: `independent_vars`
Or maybe `variables` if you don't want to touch semantics.

**Comment by Newman101** [comment]
Ok, I will amend this one. Thanks for feedback.

**Comment by Newman101** [comment]
Do I need to squash the commits, please?

**Comment by toolforger** [comment]
Yes, squashing would be the right thing to do. The intermediate commit would not interest people looking at the logs.

File: sympy/calculus/euler.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
     combinations_with_replacement, iterable, range)
 
 
-def euler_equations(L, funcs=(), vars=()):
+def euler_equations(L, funcs=(), variables=()):
```
Comment (moorepants) [inline sympy/calculus/euler.py]: If this is a public function, which it is, and you change the call signature, then it needs to be deprecated properly.

https://github.com/sympy/sympy/wiki/Deprecating-policy

**Comment by moorepants** [comment]
I'm not seeing how `vars` as a variable name was an issue. There is likely no reason to make this change. @Newman101 Please explain why it is needed.

**Comment by Newman101** [comment]
@moorepants First of all, shadowing built-ins confuses syntax highlighters in cases like Vi, Emacs, PyDev or Idle. It might also confuse some people.

**Comment by moorepants** [comment]
Confusing syntax highlighters is the least of our concerns in this case. If we break 1000 people's code because of this change, then we really have annoyed users.

You could do something like:

```python
def f(vars=()):
    # renamed here to not confuse syntax highlighters, since vars is a builtin     
    variables = vars
```

**Comment by toolforger** [comment]
Are parameter names part of the guaranteed API?

**Comment by moorepants** [comment]
> Are parameter names part of the guaranteed API?

Absolutely. It is a public declaration of function call signature. If we merge this change every bit of code that has:
```python
euler_equations(L, funcs=stuff_a, vars=stuff_b)
```
will fail.

**Comment by Newman101** [comment]
I see it now where it fails. Should have known about the declarations.

**Comment by moorepants** [comment]
My opinion is that this is not worth the hassle and we should close it. You can come up with a work around if you really want, but I'm not sure of the point. We've likely already spent enough time on this as it is.

**Comment by Newman101** [comment]
Agreed, I'll see about the workaround later but I do close this for now.

**Comment by toolforger** [comment]
Am 06.05.2015 um 21:26 schrieb Jason K. Moore:
>> Are parameter names part of the guaranteed API?
>
> Absolutely. It is a public declaration of function call signature.

Aw this sucks. I recall having seen function parameter names that 
weren't exactly well-chosen, and that's never going to get fixed because 
a marginal improvement isn't worth deprecation.
I hate this kind of bad legacy, but I guess it can't be helped.

**Comment by Newman101** [comment]
Well, in the context human reader can see which is local variable and which is built-in. How ever, syntax highlighters tend to be stupid and they make no difference between built-ins and local variables. Many programmers do actually use colour codes for different entities - for them this kind of shadow built-ins can be confusing. I'm sure I can actually build a workaround for this, but as the variable names are indeed part of guaranteed API, this solution won't work as it is.

**Pull request closed without merging**